idb/postgres/postgres_migrations.go

```diff
@@ -1936,3 +1938,104 @@ func m11AssetHoldingFrozen(db *IndexerDb, state *MigrationState) error {
 
 	return nil
 }
+
+type m12Account struct {
+	address  sdk_types.Address
+	closedAt uint64 // the round when the account was last closed
+}
+
+func m12GetAccounts(db *sql.DB) ([]m12Account, error) {
+	query := "SELECT addr, created_at FROM account WHERE closed_at IS NOT NULL AND deleted = false " +
+		"AND account_data IS NOT NULL"
+	rows, err := db.Query(query)
+	if err != nil {
+		return []m12Account{}, err
+	}
+	defer rows.Close()
+
+	var res []m12Account
+
+	for rows.Next() {
+		var addrBytes []byte
+		var closedAt sql.NullInt64
+
+		err = rows.Scan(&addrBytes, &closedAt)
+		if err != nil {
+			return []m12Account{}, err
+		}
+
+		var addr sdk_types.Address
+		copy(addr[:], addrBytes)
+		res = append(res, m12Account{addr, uint64(closedAt.Int64)})
+	}
+	if err := rows.Err(); err != nil {
+		return []m12Account{}, err
+	}
+
+	return res, nil
+}
+
+func m12FixAuthAddr(db *sql.DB, account m12Account) error {
+	// This will not work properly if the account was closed and reopened in the same round
+	// but that's unlikely to happen.
+	query := "SELECT decode(txn -> 'txn' ->> 'rekey', 'base64') FROM txn WHERE " +
+		"round > $1 AND (txn -> 'txn' ->> 'rekey') IS NOT NULL ORDER BY round DESC LIMIT 1"
```

Comment:
This probably needs to join in txn_participation, and include a filter by sender = account?